app/cafe.py

```diff
@@ -0,0 +1,21 @@
+import datetime
+from app.errors import (NotVaccinatedError,
+                        OutdatedVaccineError,
+                        NotWearingMaskError)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" not in visitor:
```

Comment:
use get() when accessing a key. fix in all places

Reply:
```python
if "vaccine" not in visitor:
```
here i just check the existence of the vaccine key
"If the visitor does not have a vaccine key, it means that he is not vaccinated. In this case, the method must raise a NotVaccinatedError exception."

why? 
```python
visitor.get("vaccine")
                .get("expiration_date")
```
is better than
```python
visitor["vaccine"]["expiration_date"]
```
cause we know for sure the vaccine key is

with 
```python
visitor.get("wearing_a_mask")
```
i fully agree

```python
kfc = Cafe("KFC")
visitor = {
    "name": "Paul",
    "age": 23,
}
kfc.visit_cafe(visitor)     # NotVaccinatedError
```
```python
import datetime

kfc = Cafe("KFC")
visitor = {
    "name": "Paul",
    "age": 23,
    "vaccine": {
        "expiration_date": datetime.date(year=2019, month=2, day=23)
    }
}
kfc.visit_cafe(visitor)     # OutdatedVaccineError
```
here i see a visitor without wearing_a_mask key can exist

Reply:
you are right `if "vaccine" not in visitor:` is fine, if you prefer like that

but you need to check if "expiration_date" exists in visitor["vaccine"]